torch_xla/experimental/megablox/tpu_features.py

```diff
@@ -0,0 +1,53 @@
+"""Module with utilities to detect TPU features."""
+# TODO(shabalin): We will need to generalize this to support GPUs.
+import re
+from absl import flags
+
+import torch
+import torch_xla
+import torch_xla.core.xla_model as xm
+from torch_xla._internal import tpu
+
+
+def is_tpu() -> bool:
+  return "TPU" in torch_xla.core.xla_model.xla_device_hw(xm.xla_device())
+
+
+def tpu_kind() -> str:
```

Comment:
we must have similar helpers somewhere in our code base. @will-cromar any idea?

Reply:
The helpers are in `torch_xla._internal.tpu`

Reply:
Sg, removed this `tpu_futures.py` file and updated to use `torch_xla._internal.tpu` directly.